BangazonAPI/Controllers/OrdersController.cs

```diff
@@ -0,0 +1,280 @@
+                            order.PaymentTypeId = reader.GetInt32(reader.GetOrdinal("PaymentTypeId"));
+                        }
+                    }
+                    reader.Close();
+                    return Ok(order);
+                }
+            }
+        }
+
+        [HttpPost]
+        public async Task<IActionResult> Post([FromBody] Order order)
+        {
+            using (SqlConnection conn = Connection)
+            {
+                conn.Open();
+                using (SqlCommand cmd = conn.CreateCommand())
+                {
+                    cmd.CommandText = "INSERT INTO [Order] (PaymentTypeId, CustomerId) OUTPUT INSERTED.Id VALUES (@PaymentTypeId, @CustomerId)";
+                    cmd.Parameters.Add(new SqlParameter("@PaymentTypeId", order.PaymentTypeId));
+                    cmd.Parameters.Add(new SqlParameter("@CustomerId", order.CustomerId));
+
+
+                    int newId = (int)cmd.ExecuteScalar();
+                    order.Id = newId;
+                    return CreatedAtRoute("GetOrder", new { id = newId }, order);
+                }
+            }
+        }
+
+
+
+        [HttpPut("{id}")]
+        public async Task<IActionResult> Put([FromRoute] int id, [FromBody] Order order)
+        {
+            try
+            {
+                using (SqlConnection conn = Connection)
+                {
+                    conn.Open();
+                    using (SqlCommand cmd = conn.CreateCommand())
+                    {
+                        cmd.CommandText = @"UPDATE [Order] SET PaymentTypeId = @PaymentTypeId, CustomerId = @CustomerId WHERE Id = @id";
+
+
+                        cmd.Parameters.Add(new SqlParameter("@PaymentTypeId", order.PaymentTypeId));
+                        cmd.Parameters.Add(new SqlParameter("@CustomerId", order.CustomerId));
+                        cmd.Parameters.Add(new SqlParameter("@id", id));
+
+                        int rowsAffected = cmd.ExecuteNonQuery();
+                        if (rowsAffected > 0)
+                        {
+                            return new StatusCodeResult(StatusCodes.Status204NoContent);
+                        }
+                        throw new Exception("No rows affected");
+                    }
+                }
+            }
+            catch (Exception)
+            {
+                if (!OrderExists(id))
+                {
+                    return NotFound();
+                }
+                else
+                {
+                    throw;
+                }
+            }
+        }
+
+        [HttpDelete("{id}")]
+        public async Task<IActionResult> Delete([FromRoute] int id)
+        {
+            try
+            {
+                using (SqlConnection conn = Connection)
+                {
+                    conn.Open();
+                    using (SqlCommand cmd = conn.CreateCommand())
+                    {
+                        cmd.CommandText = @"DELETE FROM [Order] WHERE ID = @id";
```

Comment:
So, your test passes! Hooray! BUT, I actually think that's a false positive. In the test, you're deleting an order that has no products associated with it (via the join table). 

When I tried to delete an order in Postman, I got a 500 error because of the foreign key restraint on the ProductOrder join table.

The rest of your group didn't run into this problem because they used soft delete. For consistency's sake, and for the sake of record-keeping, you should use soft delete too. (But, just FYI: this is why we use soft delete! It causes all kinds of problems when you try to delete something with FKs attached to it.)